Code/src/main/java/graph/Node.java

```diff
@@ -1,13 +1,34 @@
 package graph;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 public class Node<T> {
 	private int id;
 	private T value;
-	private List<Node> neighbours;
+	private final List<Node> neighbours;
 	private int weightStorage;
 
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+		Node<?> node = (Node<?>) o;
+		return id == node.id &&
+				weightStorage == node.weightStorage &&
+				value.equals(node.value);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(id, value, weightStorage);
+	}
+
+	{
+		neighbours = new ArrayList<>();
+	}
```

Comment:
Dlaczego przypisanie neighbours jako ArrayList znajduje się w bloku kodu nieprzypisanym do żadnej metody?

Reply:
Bo są dwa konstruktory. To jest blok inicjalizacyjny. Wykonuje się przed dowolnym konstruktorem